libs/designsystem/accordion/src/accordion.directive.ts

```diff
@@ -1,8 +1,28 @@
-import { Directive } from '@angular/core';
+import { AfterContentInit, ContentChildren, Directive, Input, QueryList } from '@angular/core';
+
+import { startWith } from 'rxjs';
+import { AccordionItemComponent } from './accordion-item.component';
 
 @Directive({
   selector: '[kirbyAccordion], kirby-accordion',
   exportAs: 'kirby-accordion',
   standalone: false,
 })
-export class AccordionDirective {}
+export class AccordionDirective implements AfterContentInit {
+  @Input() headingLevel: number;
+  @ContentChildren(AccordionItemComponent) accordionItems: QueryList<AccordionItemComponent>;
+
+  ngAfterContentInit(): void {
+    this.accordionItems.changes.pipe(startWith(0)).subscribe(() => {
+      this.setHeadingLevels();
+    });
+  }
+
+  private setHeadingLevels(): void {
+    if (this.accordionItems && this.headingLevel) {
```

Comment:
Did you see my additional comment here?

https://github.com/kirbydesign/designsystem/pull/3974#discussion_r2182709698